stubs/PyAutoGUI/pyautogui/__init__.pyi

```diff
@@ -0,0 +1,233 @@
+import collections.abc
+from _typeshed import Incomplete
+from collections.abc import Callable, Generator
+from datetime import datetime
+from typing import NamedTuple, TypeVar, overload
+from typing_extensions import ParamSpec
+
+# from pyscreeze import Box
+
+class PyAutoGUIException(Exception): ...
+class FailSafeException(PyAutoGUIException): ...
+class ImageNotFoundException(PyAutoGUIException): ...
+
+collectionsSequence = collections.abc.Sequence
+
+_P = ParamSpec("_P")
+_R = TypeVar("_R")
+
+# TODO: Complete types with pyscreeze once we can import non-types dependencies
+# See: https://github.com/python/typeshed/issues/5768
+
+class _Box(NamedTuple):
+    left: int
+    top: int
+    width: int
+    height: int
+
+def raisePyAutoGUIImageNotFoundException(wrappedFunction: Callable[_P, _R]) -> Callable[_P, _R]: ...
+def locate(*args: Incomplete, **kwargs: Incomplete) -> _Box | None: ...
+def locateAll(*args: Incomplete, **kwargs: Incomplete) -> Generator[_Box, None, None]: ...
+def locateAllOnScreen(*args: Incomplete, **kwargs: Incomplete) -> Generator[_Box, None, None]: ...
+def locateCenterOnScreen(*args: Incomplete, **kwargs: Incomplete) -> Point | None: ...
+def locateOnScreen(*args: Incomplete, **kwargs: Incomplete) -> _Box | None: ...
+def locateOnWindow(*args: Incomplete, **kwargs: Incomplete) -> _Box | None: ...
```

Comment:
For parameters where we don't know (or haven't yet figured out) what the annotations should be, we prefer to leave them unannotated rather than using `_typeshed.Incomplete`. See #8472 for some recent discussion as to why.

(This only applies for parameter annotations where the type is completely unknown. For some types, we might know that the type "could be `None`", and in that situation, we might use `Incomplete | None`. But for completely unknown types, we like to leave it unannotated.)

```suggestion
def locate(*args, **kwargs) -> _Box | None: ...
def locateAll(*args, **kwargs) -> Generator[_Box, None, None]: ...
def locateAllOnScreen(*args, **kwargs) -> Generator[_Box, None, None]: ...
def locateCenterOnScreen(*args, **kwargs) -> Point | None: ...
def locateOnScreen(*args, **kwargs) -> _Box | None: ...
def locateOnWindow(*args, **kwargs) -> _Box | None: ...
```